tldr/cross_file_calls.py

```
@@ -3939,3 +3965,1016 @@ def _build_php_call_graph(
+        # Fallback: walk descendants for first identifier-like
+        for child in call_node.children:
+            for sub in child.children:
+                if sub.type in name_child_types:
+                    first_text = source[sub.start_byte:sub.end_byte].decode("utf-8", errors="replace")
+                    return ("direct", first_text)
+        return None
+
+    def extract_calls_in(body_node) -> list[tuple[str, str]]:
+        calls = []
+
+        def visit(node):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `visit`

Add return type annotation: `None`

(ANN202)